kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/KeyManager.java

```diff
@@ -24,8 +24,8 @@ public interface KeyManager<K> {
      * Asynchronously encrypt the given {@code recordRequests} using the current DEK for the given KEK, calling the given receiver for each encrypted record
      * @param encryptionScheme The encryption scheme.
      * @param records The requests to be encrypted.
-     * @param receiver The receiver of the encrypted buffers.
-     * @return A completion stage that completes when all the records in the given {@code recordRequests} have been processed.
+     * @param receiver The receiver of the encrypted buffers. The receiver will recieve the buffers in the original order and can expect no parallel calls.
```

Comment:
I'm trying to think of a concise way to say it. The `Receiver` implementation can expect that it will be delivered (record, bufferWithEncryptedValue, headers) in the same order as the records in the RecordBatch. Network order might be the right wording?

Reply:
actually it's not really the record batch, it's the same order as the `List<Records>` passed to `KeyManager#encrypt`

Reply:
currently the EnvelopeEncryptionFilter is built on that assumption:

```
return keyManager.encrypt(
                                    new EncryptionScheme<>(kekId, EnumSet.of(RecordField.RECORD_VALUE)),
                                    encryptionRequests,
                                    (kafkaRecord, encryptedValue, headers) -> builder.append(kafkaRecord.timestamp(), kafkaRecord.key(), encryptedValue, headers)) // expects MemoryRecords builder to have data appended in the same order as originally received
                                    .thenApply(ignored -> ppd.setRecords(builder.build()));
                                    ```

Reply:
So I figured we should write it down somewhere if `KeyManagers` are supposed to feed the `Receiver` records in a particular order.

Reply:
Maybe the phrase is something like `batch iteration order` or `record iteration order`?

Reply:
I don't think talking about the batches is right.  KeyManager has no knowledge of batches, only a list of records.  I think we just want to say something like: `The receiver of the encrypted buffers.  The receiver is guaranteed to receive the encrypted buffers sequentially, in the same order as {@code records}, with no possibility of parallel invocation`